src/main/java/io/boomerang/eventing/nats/jetstream/TinCanCommunication.java

```diff
@@ -0,0 +1,233 @@
+package io.boomerang.eventing.nats.jetstream;
+
+import java.io.IOException;
+import java.lang.ref.Reference;
+import java.lang.ref.WeakReference;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import io.boomerang.eventing.nats.ConnectionPrimer;
+import io.boomerang.eventing.nats.ConnectionPrimerListener;
+import io.boomerang.eventing.nats.jetstream.exception.ConsumerNotFoundException;
+import io.boomerang.eventing.nats.jetstream.exception.NoNatsConnectionException;
+import io.boomerang.eventing.nats.jetstream.exception.StreamNotFoundException;
+import io.boomerang.eventing.nats.jetstream.exception.SubjectMismatchException;
+import io.nats.client.Connection;
+import io.nats.client.Dispatcher;
+import io.nats.client.JetStreamApiException;
+import io.nats.client.JetStreamSubscription;
+import io.nats.client.Message;
+import io.nats.client.MessageHandler;
+import io.nats.client.PullSubscribeOptions;
+import io.nats.client.PushSubscribeOptions;
+import io.nats.client.api.ConsumerConfiguration;
+import io.nats.client.api.ConsumerInfo;
+import io.nats.client.api.PublishAck;
+import io.nats.client.api.StreamConfiguration;
+import io.nats.client.api.StreamInfo;
+import io.nats.client.impl.NatsMessage;
+
+public class TinCanCommunication implements ConnectionPrimerListener {
```

Comment:
What do we mean by TinCan? Im not following the reason for this enhancement?

Reply:
So i think I understand its purpose. The TinCanCommunication and Primer is mainly used for the Listener -> Workflow example? i.e. 1:1 or am I misunderstanding its purpose?